### ISSUE TITLE
[BUG] Limited mouse movement

### ISSUE THREAD

**Issue by spectrum705** [hmlendea/gfn-electron#250]
### Describe the bug

I am noticing limited / locked mouse movement across both axis. idk why it happens. I have the latest client.
I am using ubuntu 

### To Reproduce

I dont really no the way but i was playing genshin impact

### Operating System

ubuntu

### Desktop Environment

GNONE 42.9

### Display Server

wayland i think

### Installation method

flatpak

### Version

2.1.2  

### Is this a fresh install of the app or an update from a past version?

update

### Did this issue appear right away upon installation/updating, or spontaneously?

all the time, even previously

**Comment by unixbird** [comment]
I actually experience the mouse not being fully captured as well if you have multiple monitors. Not sure if this is due to using Wayland or not.

**Comment by spectrum705** [comment]
I only have single monitor tho

**Comment by Freaksed** [comment]
i have the same issue. i am on garuda (arch) installed from flatpak

**Comment by spectrum705** [comment]
yeah, another note. pressing escape unbinds the mouse. esc is need for closing menu and a lot of things in game. can we do something about this?

**Comment by Monoxide8969** [comment]
This happens to me too while playing genshin (single monitor)

**Comment by spectrum705** [comment]
it works smoothly on x11?

**Comment by Monoxide8969** [comment]
It does on my machine

**Comment by MGM175** [comment]
Also managed to replicate this error. It's for 100% related to the Wayland.

**Comment by spectrum705** [comment]
so i can play if i log in with x11 right guys?

**Comment by unixbird** [comment]
> so i can play if i log in with x11 right guys?

Correct. I tested this out just now actually and it worked perfectly on Plasma 6 X11.

**Comment by spectrum705** [comment]
oh okay

**Comment by GeorgelPreput** [comment]
Also got it working under Wayland using Gamescope:

```
gamescope -b -W 1920 -H 1080 -r 60 --backend sdl --expose-wayland --prefer-vk-device --force-grab-cursor -- geforcenow
```

Probably not all parameters are needed, but this is what worked for me. Should note though that this only worked with the AUR version, and not with the Flatpak. Would be pretty cool though to integrate `gamescope` into the Flatpak, since it also brings upscaling. GeForce Now streaming only goes as high as 1080p

**Comment by RickMars-Tech** [comment]
Apparently this bug does not happen when using the Appimage version, while they solve it they can choose to use this option instead of using Flatpak, (btw using the "Nix Version" is also not an option since there is the same bug)

**Comment by Play-InTheWay** [comment]
In addition to running it on an x11 session, running it on Xwayland also does fix it; that is why it works on AppImage because it runs with Xwayland by default.

Noteworthy is that when I try to run GeForce Now through Brave in native Wayland, the error is also fixed. So it could be fixable under native Wayland, but until then, the Flatpak version shouldn't enable Wayland by default.

**Comment by spectrum705** [comment]
> Apparently this bug does not happen when using the Appimage version, while they solve it they can choose to use this option instead of using Flatpak, (btw using the "Nix Version" is also not an option since there is the same bug)

i am having trouble running the app image version on ubuntu

**Comment by Kaympe20** [comment]
Same issue

**Comment by Play-InTheWay** [comment]
You all can fix it by running it on Xwayland; do it by adding this to the executable command: "XDG_SESSION_TYPE=x11"

**Comment by spectrum705** [comment]
> You all can fix it by running it on Xwayland; do it by adding this to the executable command: "XDG_SESSION_TYPE=x11"

i open the application directly by clicking the app image file, where should i add this?

**Comment by Play-InTheWay** [comment]
> > You all can fix it by running it on Xwayland; do it by adding this to the executable command: "XDG_SESSION_TYPE=x11"
> 
> i open the application directly by clicking the app image file, where should i add this?

Open a terminal in the same folder where the AppImage is and run `XDG_SESSION_TYPE=x11 ./geforcenow-electron_2.1.3_linux.AppImage `
Though in general for managing AppImages, I recommend using something like Gear Lever, which automatically creates .desktops and allows easy access and modifications.

**Comment by spectrum705** [comment]
> > > You all can fix it by running it on Xwayland; do it by adding this to the executable command: "XDG_SESSION_TYPE=x11"
> > 
> > 
> > i open the application directly by clicking the app image file, where should i add this?
> 
> Open a terminal in the same folder where the AppImage is and run `XDG_SESSION_TYPE=x11 ./geforcenow-electron_2.1.3_linux.AppImage ` Though in general for managing AppImages, I recommend using something like Gear Lever, which automatically creates .desktops and allows easy access and modifications.

when i ran it i got this error Running as root without --no-sandbox is not supported. 
and when i added --no-sandbox tag , gfn window did open but it was jusst stuck on white screen

**Comment by Play-InTheWay** [comment]
> > > > You all can fix it by running it on Xwayland; do it by adding this to the executable command: "XDG_SESSION_TYPE=x11"
> > > 
> > > 
> > > i open the application directly by clicking the app image file, where should i add this?
> > 
> > 
> > Open a terminal in the same folder where the AppImage is and run `XDG_SESSION_TYPE=x11 ./geforcenow-electron_2.1.3_linux.AppImage ` Though in general for managing AppImages, I recommend using something like Gear Lever, which automatically creates .desktops and allows easy access and modifications.
> 
> when i ran it i got this error Running as root without --no-sandbox is not supported. and when i added --no-sandbox tag , gfn window did open but it was jusst stuck on white screen

Don't run it as root.

**Comment by spectrum705** [comment]
i upgraded ubuntu, that seems to be running x11 by default. it runs better now
hey @Play-InTheWay do you know how to exit tho? i press Esc but that doest exist gfn.

**Comment by Play-InTheWay** [comment]
> i upgraded ubuntu, that seems to be running x11 by default. it runs better now
hey @Play-InTheWay do you know how to exit tho? i press Esc but that doest exist gfn.

I can do it without issue. I don't know how you could fix it.

**Comment by spectrum705** [comment]
even when i hold Esc key, nothing seems to happen

**Comment by realestninja** [comment]
> You all can fix it by running it on Xwayland; do it by adding this to the executable command: "XDG_SESSION_TYPE=x11"

This really does the trick for me! 🥳


> even when i hold Esc key, nothing seems to happen

@spectrum705  For exiting, have you tried Alt+F4? From what I can see, it has been added as a binding: https://github.com/hmlendea/gfn-electron/blob/427bd0c7312630fb62df732848a046423cd8e06a/scripts/main.js#L114

**Comment by spectrum705** [comment]
@realestninja  yeah i tried Alt + F4 too. didnt seem to close it

**Comment by hmlendea** [comment]
Please re-open if this issue still affects the latest version of this app.

**Comment by ilikecatgirls** [comment]
It seems to be affecting me as well at the moment, using the latest version, on arch linux kde, running wayland. its essentially like the initial comment described it. i can somewhat fix it by pointing towards the top right to somewhat reset the mouse area, but it happens again after moving it a bit. on games that lock the mouse, to rotate etc, I can only rotate to a certain extent.

Edit: after reading the README a bit I had to include the fact that i am using the AUR version of the package.
